src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,332 @@
+    {
+        if (analyzerResult is null)
+        {
+            throw new ArgumentNullException(nameof(analyzerResult));
+        }
+        if (workspace is null)
+        {
+            throw new ArgumentNullException(nameof(workspace));
+        }
+
+        ProjectId projectId = ProjectId.CreateFromSerialized(analyzerResult.ProjectGuid);
+
+        ConcurrentDictionary<Guid, string[]> WorkspaceProjectReferences = new ConcurrentDictionary<Guid, string[]>();
+        WorkspaceProjectReferences[projectId.Id] = analyzerResult.ProjectReferences.ToArray();
+
+        ProjectInfo projectInfo = GetProjectInfo(analyzerResult, workspace, projectId);
+        if (projectInfo is null)
+        {
+            // Error
+            return null;
+        }
+        Solution solution = workspace.CurrentSolution.AddProject(projectInfo);
+
+        if (!workspace.TryApplyChanges(solution))
+        {
+            throw new InvalidOperationException("Could not apply workspace solution changes");
+        }
+
+        return workspace.CurrentSolution.GetProject(projectId);
+    }
+
+    internal static AdhocWorkspace CreateWorkspace()
+    {
+        AdhocWorkspace workspace = new AdhocWorkspace();
+        workspace.WorkspaceChanged += WorkspaceChangedHandler;
+        workspace.WorkspaceFailed += WorkspaceFailedHandler;
+        return workspace;
+    }
+
+    private static void WorkspaceFailedHandler(object sender, WorkspaceDiagnosticEventArgs e)
+    {
+        // Log error
+    }
+
+    private static void WorkspaceChangedHandler(object sender, WorkspaceChangeEventArgs e)
+    {
+        // Log
+    }
+
+    private static ProjectInfo GetProjectInfo(BuildProjectData buildProjectData, CodeAnalysis.Workspace workspace, ProjectId projectId)
+    {
+        string projectName = Path.GetFileNameWithoutExtension(buildProjectData.ProjectFilePath);
+        if (!TryGetSupportedLanguageName(buildProjectData.ProjectFilePath, out string languageName))
+        {
+            return null;
+        }
+        return ProjectInfo.Create(
+            projectId,
+            VersionStamp.Create(),
+            projectName,
+            projectName,
+            languageName,
+            filePath: buildProjectData.ProjectFilePath,
+            outputFilePath: buildProjectData.TargetPath,
+            documents: GetDocuments(buildProjectData, projectId),
+            projectReferences: GetExistingProjectReferences(buildProjectData, workspace),
+            metadataReferences: GetMetadataReferences(buildProjectData),
+            analyzerReferences: GetAnalyzerReferences(buildProjectData, workspace),
+            parseOptions: CreateParseOptions(buildProjectData, languageName),
+            compilationOptions: CreateCompilationOptions(buildProjectData, languageName));
+    }
+
+    private static bool TryGetSupportedLanguageName(string projectPath, out string languageName)
```

Comment:
Is there a reason for this method to exist given that C# is the only language this project supports?

Reply:
removed